seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/common/AbstractCatalogSupportTransform.java

```diff
@@ -20,18 +20,36 @@
 import org.apache.seatunnel.api.table.catalog.CatalogTable;
 import org.apache.seatunnel.api.table.catalog.TableIdentifier;
 import org.apache.seatunnel.api.table.catalog.TableSchema;
+import org.apache.seatunnel.api.table.type.SeaTunnelRow;
+import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+import org.apache.seatunnel.api.transform.SeaTunnelTransform;
 
 import lombok.NonNull;
 
-public abstract class AbstractCatalogSupportTransform extends AbstractSeaTunnelTransform {
+public abstract class AbstractCatalogSupportTransform implements SeaTunnelTransform<SeaTunnelRow> {
     protected CatalogTable inputCatalogTable;
 
     protected volatile CatalogTable outputCatalogTable;
 
+    protected String inputTableName;
+    protected SeaTunnelRowType inputRowType;
```

Comment:
1. Please only keep inputTableName.
2. We can move it to SQLTransform because only it use this field now.

Reply:
done